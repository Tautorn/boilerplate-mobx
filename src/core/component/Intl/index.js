import React, { Component, PropTypes } from 'react'
import { intlShape, injectIntl } from 'react-intl'

class Intl extends Component {

  getMessage(id, values) {
    const message = this.props.intl.messages[id]

    return message ? this.props.intl.formatMessage({ id }, values) : id
  }

  render() {
    const { id, values } = this.props

    return (
      <span>
        {this.getMessage(id, values)}
      </span>
    )
  }
}

Intl.propTypes = {
  id: PropTypes.string,
  intl: intlShape.isRequired,
  values: PropTypes.object
}

export default injectIntl(Intl)
