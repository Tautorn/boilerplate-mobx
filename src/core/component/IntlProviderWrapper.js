import React, { PropTypes } from 'react'
import { IntlProvider } from 'react-intl'
import Translate from 'core/component/translate'

const language = Translate.language
const messages = Translate.messages

const IntlProviderWrapper = ({ children }) => (
  <IntlProvider locale={language} messages={messages}>
    {children}
  </IntlProvider>
)

IntlProviderWrapper.propTypes = {
  children: PropTypes.node.isRequired
}

export default IntlProviderWrapper
