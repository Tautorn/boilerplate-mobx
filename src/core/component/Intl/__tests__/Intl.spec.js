import React from 'react'
import { render } from 'enzyme'
import IntlProviderWrapper from '../../IntlProviderWrapper'
import Intl from '../'

describe('Intl', () => {

  it('should render Intl component', () => {
    const intl = render(
      <IntlProviderWrapper>
        <Intl id="hello" values={{ name: 'hi' }} />
      </IntlProviderWrapper>
    )

    expect(intl).toMatchSnapshot()
  })

  it('should render fallback when intl is not found', () => {
    const intl = render(
      <IntlProviderWrapper>
        <Intl id="errrorr" />
      </IntlProviderWrapper>
    )

    expect(intl).toMatchSnapshot()
  })
})
