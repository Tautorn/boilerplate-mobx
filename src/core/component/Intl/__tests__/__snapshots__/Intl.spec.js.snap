// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Intl should render Intl component 1`] = `
<span>
  hello
</span>
`;

exports[`Intl should render fallback when intl is not found 1`] = `
<span>
  errrorr
</span>
`;
